GoMeet/src/main/webapp/js/meeting-display.js

```diff
@@ -1,13 +1,28 @@
 /**
  * Toggles meeting event display depending on login status of user.
  */
-function toggleMeetingDisplay() {
-  LoginStatus.doGet().then((loginStatus) => {
+async function toggleMeetingDisplay() {
+  let meetingEventId = getMeetingEventId();
+  await LoginStatus.doGet(meetingEventId).then((loginStatus) => {
     let prompt; 
     let style; 
     if (loginStatus.loggedIn === 'true') {
-      style = 'block'; 
       prompt = '<a href=' + loginStatus.logoutUrl + '>Logout</a>';
+      // Check if the meeting event ID is valid (i.e. there exists a MeetingEvent entity 
+      // in Datastore with that key)
+      let meetingEventEntity = MeetingEventDAO.fetchMeetingEvent(meetingEventId); 
+      if (meetingEventEntity['status'] === undefined) { // No error response was sent 
+        style = 'block';
+        let meetingName = meetingEventEntity['meetingName'];
+        let durationMins = meetingEventEntity['durationMins'];
+        let durationHours = meetingEventEntity['durationHours']; 
+        let timeFindMethod = meetingEventEntity['timeFindMethod']; 
+        let guestList = meetingEventEntity['guestList']; 
+        let meetingTimeIds = meetingEventEntity['meetingTimeIds']; 
+        let meetingLocationIds = meetingEventEntity['meetingLocationIds'];
+        // TO DO: Call neccessary functions to display meeting event data
+        displayMeetingTimeForm(); 
+      }
```

Comment:
The rationale behind this implementation design is that the user should be able to log in even if they are on a meeting event page with a query string entity key that does not exist. After logging in, the application will verify that a meeting event entity with the given ID exists in datastore and if so, it will call the necessary functions to display the data.